hayeon/programmers-1/하샤드-수.js

```diff
@@ -0,0 +1,22 @@
+function solution(x) {
+  const newArr = x
+    .toString()
+    .split("")
+    .map((arg) => Number(arg));
+  let sum = 0;
+
+  for (let i = 0; i < newArr.length; i++) {
+    sum += newArr[i];
+  }
+  return x % sum === 0 ? true : false;
+}
```

Comment:
```suggestion
function solution(x) {
  const newArr = x
    .toString()
    .split("")
    .map(Number);
  let sum = 0;

  for (let i = 0; i < newArr.length; i++) {
    sum += newArr[i];
  }
  return x % sum === 0 ? true : false;
}
```
- 잘푸셨습니다 👍 map도 좋은 방법이지만 아래와 같이 reduce 메서드를 활용하면 좀 더 깔끔한 풀이가 가능합니다 참고해주세요

```js
function solution(x) {
  const digitSum = String(x)
    .split("")
    .reduce((acc, cur) => acc + Number(cur), 0);

  return x % digitSum === 0 ? true : false;
}
```

Reply:
reduce 함수는 예전에 살짝 이해하고 정리만 해놨었는데 이렇게 쓰일 수도 있었군요,, reduce이용해서 다시 풀어보겠슴닷!! 항상 감사합니다ㅠㅠ 다양한 코드 보니까 여러가지 방식으로 풀 수 있다는 것도 체감이되네용,,,호오👍🏻